src/components/watchers-form.vue

```diff
@@ -329,7 +356,20 @@ export default {
         }
       }
 
-      if (valid ) {
+      if (
+        !this.config.watching ||
+        !this.config.script ||
+        !this.jsonIsValid('input_data') ||
+        !this.jsonIsValid('script_configuration')
+      ) {
+        valid = false;
+      }
+
+      if (!valid && globalObject.ProcessMaker && globalObject.ProcessMaker.alert) {
+        globalObject.ProcessMaker.alert(this.$t('An error occurred. Check the form for errors in red text.'), 'danger');
+      }
+
+      if (valid) {
```

Comment:
I think this method can be refactored a bit. The `valid` variable it set in two places, and I think the logic for that could be broken out into a separate method, making it easier to use an early return. I'm thinking of something like this:

```js
if (this.isFormValid()) {
  // show alert
  return;
}

this.save();
```

A name like `validateDataAndSave` instead of `validateData` may also better communicate what this function is doing.

Reply:
I like the idea

Reply:
@chipit24 the changes have already been made.